packages/fastify/src/withClerkMiddleware.ts

```diff
@@ -1,14 +1,16 @@
-import type { ClerkOptions } from '@clerk/types';
 import { parse } from 'cookie';
 import type { FastifyReply, FastifyRequest } from 'fastify';
 
 import { clerkClient } from './clerkClient';
 import * as constants from './constants';
+import type { ClerkFastifyOptions } from './types';
 import { getSingleValueFromArrayHeader } from './utils';
 
-export const withClerkMiddleware = (options: ClerkOptions) => {
+export const withClerkMiddleware = (options: ClerkFastifyOptions) => {
   return async (req: FastifyRequest, reply: FastifyReply) => {
     const cookies = parse(req.raw.headers.cookie || '');
+    const secretKey = options.secretKey || constants.SECRET_KEY;
+    const publishableKey = options.publishableKey || constants.PUBLISHABLE_KEY;
```

Comment:
❓ should we create a util method in backend to load
```
/*
 * prefix: should end with a `_` if provided
 */
const loadOptionsFromEnv = ( prefix?: string = '' ): Record<string, string | undefined> => {
  return {
   secretKey: process.env[prefix + 'CLERK_API_KEY],
   publishableKey: process.env[prefix + 'CLERK_PUBLISHABLE_KEY],
  }
}

// example calling it from remix
const envOptions = loadFromEnv();
// example calling it from next
const envOptions = loadFromEnv('NEXT_PUBLIC_');
```
and merge it with options like this?
```
authenticateRequest({
  ...envOptions,
  ...options,
  //...rest
})
```

Reply:
Definitely, 100% agree.
A layer to abstract this is necessary. I was thinking about a mechanism of some kind that would act as an adapter, eg: accept a method that reads the any config we need in a framework-specific way (prefixes might not be enough) and output a framework-agnostic config object we will use internally.

WDYT?

Cc @SokratisVidros